frontend/src/app/plans/edit/[id]/_components/SyncErrorAlert.tsx

```diff
@@ -0,0 +1,137 @@
+"use client";
+
+import { formatDistance, isAfter, isEqual } from "date-fns";
+import { pl } from "date-fns/locale";
+import { DownloadCloudIcon, Loader2Icon, UploadCloudIcon } from "lucide-react";
+import React from "react";
+
+import { Button } from "@/components/ui/button";
+import { cn } from "@/lib/utils";
+
+interface PlanResponseType {
+  name: string;
+  userId: number;
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  courses: Array<{
+    id: string;
+    name: string;
+    department: string;
+    lecturer: string;
+    type: string;
+    ects: number;
+    semester: number;
+    groups: Array<{
+      id: number;
+      name: string;
+      day: string;
+      time: string;
+      room: string;
+    }>;
+  }>;
+  registrations: Array<{
+    id: string;
+    name: string;
+  }>;
+}
+
+export const SyncErrorAlert = ({
+  onlinePlan,
+  planDate,
+  bounce = false,
+  downloadChanges,
+  sendChanges,
+}: {
+  onlinePlan: PlanResponseType | null | undefined;
+  planDate: Date;
+  bounce?: boolean;
+  downloadChanges: () => void;
+  sendChanges: () => void;
+}) => {
+  const [loadingSending, setLoadingSending] = React.useState(false);
```

Comment:
zamień na useState zwykły wszędzie bo mieszasz
```suggestion
  const [loadingSending, setLoadingSending] = useState(false);
```

Reply:
good